extension/WebJobs.Extensions.RabbitMQ/Trigger/RabbitMQListener.cs

```diff
@@ -152,6 +158,30 @@ public ScaleStatus GetScaleStatus(ScaleStatusContext<RabbitMQTriggerMetrics> con
             return this.GetScaleStatusCore(context.WorkerCount, context.Metrics?.ToArray());
         }
 
+        internal static Activity StartActivity(BasicDeliverEventArgs ea)
+        {
+            Activity activity;
+            if (ea.BasicProperties.Headers != null && ea.BasicProperties.Headers.ContainsKey("traceparent"))
+            {
+                byte[] traceParentIdInBytes = ea.BasicProperties.Headers["traceparent"] as byte[];
+                string traceparentId = Encoding.Default.GetString(traceParentIdInBytes);
+                activity = ActivitySource.StartActivity("Trigger", ActivityKind.Consumer, traceparentId);
+            }
+            else
+            {
+                activity = ActivitySource.StartActivity("Trigger", ActivityKind.Server);
+                if (ea.BasicProperties.Headers == null)
```

Comment:
Can you check if the [null-coalescing assignment operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator) from C# 8.0 works here?

```suggestion
ea.BasicProperties.Headers ??= new Dictionary<string, object>();
```